fuel-tests/tests/contract.rs

```diff
@@ -44,3 +45,38 @@ async fn test_contract_balance() {
         assert_eq!(balance, test_bal);
     }
 }
+
+#[rstest]
+#[case(PageDirection::Forward)]
+#[case(PageDirection::Backward)]
+#[tokio::test]
+async fn test_first_5_contract_balances(#[case] direction: PageDirection) {
```

Comment:
instead of testing only the "first 5", could we parameterize the number of balances, the start cursor and the page size? That way we can also test whether things like has_previous_page or has_next_page work correctly, or that the number of results returned is always correct etc

Reply:
Alternately, maybe we can file a tech debt issue to refactor all pagination logic into a common function across all APIs and then we only need these kinds of tests in one place instead of every api that has pagination logic.

Reply:
https://github.com/FuelLabs/fuel-core/issues/382